File: Tests/Extensions/ChopTests.fs
namespace Tests.Extensions

open System;
open Stringier.Extensions;
open Xunit;

module ChopTests =
    [<Fact>]
    let Chop0 () =
        Assert.Throws(typeof<ArgumentOutOfRangeException>, fun () -> chop 0 "hello world" |> ignore);

    [<Fact>]
    let Chop1 () =
        Assert.Equal<seq<string>>([|"h";"e";"l";"l";"o";" ";"w";"o";"r";"l";"d"|], chop 1 "hello world")

    [<Fact>]
    let Chop2 () =
        Assert.Equal<seq<string>>([|"he";"ll";"o ";"wo";"rl";"d"|], chop 2 "hello world");

    [<Fact>]
    let Chop3 () =
        Assert.Equal<seq<string>>([|"hel";"lo ";"wor";"ld"|], chop 3 "hello world");

    [<Fact>]
    let Chop4 () =
        Assert.Equal<seq<string>>([|"hell";"o wo";"rld"|], chop 4 "hello world");

    [<Fact>]
    let Chop5 () =
        Assert.Equal<seq<string>>([|"hello";" worl";"d"|], chop 5 "hello world");

    [<Fact>]
    let Chop6 () =
        Assert.Equal<seq<string>>([|"hello ";"world"|], chop 6 "hello world");

    [<Fact>]
    let Chop7 () =
        Assert.Equal<seq<string>>([|"hello w";"orld"|], chop 7 "hello world");

    [<Fact>]
    let Chop8 () =
        Assert.Equal<seq<string>>([|"hello wo";"rld"|], chop 8 "hello world");

    [<Fact>]
    let Chop9 () =
        Assert.Equal<seq<string>>([|"hello wor";"ld"|], chop 9 "hello world");

    [<Fact>]
    let Chop10 () =
        Assert.Equal<seq<string>>([|"hello worl";"d"|], chop 10 "hello world");

    [<Fact>]
    let Chop11 () =
        Assert.Equal<seq<string>>([|"hello world"|], chop 11 "hello world");

    [<Fact>]
    let Chop12 () =
        Assert.Equal<seq<string>>([|"hello world"|], chop 12 "hello world");

File: Tests/ChopTests.cs
using System;
using Stringier;
using Xunit;

namespace Tests {
	public class ChopTests {
		[Theory]
		[InlineData("hello world", 1, new[] { "h", "e", "l", "l", "o", " ", "w", "o", "r", "l", "d" })]
		[InlineData("hello world", 2, new[] { "he", "ll", "o ", "wo", "rl", "d" })]
		[InlineData("hello world", 3, new[] { "hel", "lo ", "wor", "ld" })]
		[InlineData("hello world", 4, new[] { "hell", "o wo", "rld" })]
		[InlineData("hello world", 5, new[] { "hello", " worl", "d" })]
		[InlineData("hello world", 6, new[] { "hello ", "world" })]
		[InlineData("hello world", 7, new[] { "hello w", "orld" })]
		[InlineData("hello world", 8, new[] { "hello wo", "rld" })]
		[InlineData("hello world", 9, new[] { "hello wor", "ld" })]
		[InlineData("hello world", 10, new[] { "hello worl", "d" })]
		[InlineData("hello world", 11, new[] { "hello world" })]
		[InlineData("hello world", 12, new[] { "hello world" })]
		public void Chop(String text, Int32 size, String[] expected) => Assert.Equal(expected, text.Chop(size));

		[Fact]
		public void Chop_Invalid() => Assert.Throws<ArgumentOutOfRangeException>(() => "hello world".Chop(0));
	}
}
